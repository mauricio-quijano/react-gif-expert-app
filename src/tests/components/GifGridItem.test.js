import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Tests on <GifGridItem /> component', () => {
    const image = {
        title: 'Image Title',
        url: '127.0.0.1',
    };
    const wrapper = shallow(<GifGridItem {...image} />);

    test('Should display component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should contain a paragraph with the image title', () => {
        const pTag = wrapper.find('p');
        expect(pTag.text().trim()).toBe(image.title);
    });

    test('should contain an image with url as source and title as alt', () => {
        const imgTag = wrapper.find('img');
        expect(imgTag.prop('src')).toBe(image.url);
        expect(imgTag.prop('alt')).toBe(image.title);
    });

    test('should contain a div with class animate__fadeIn', () => {
        const divTag = wrapper.find('div');
        expect(divTag.prop('className').includes('animate__fadeIn')).toBe(true);
    });
});
