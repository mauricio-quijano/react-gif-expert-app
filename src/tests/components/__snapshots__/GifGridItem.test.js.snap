// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifGridItem /> component Should display component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Image Title"
    src="127.0.0.1"
  />
  <p>
    Image Title
  </p>
</div>
`;
